orttraining/orttraining/training_ops/cpu/nn/dropout_op.cc

```diff
@@ -75,16 +80,20 @@ Status Dropout<T1, T2>::Compute(OpKernelContext* context) const {
     return gsl::make_span(temp_mask_buffer.get(), X_shape.Size());
   }();
 
-  ORT_ENFORCE(Y->Shape() == X_shape, "X and Y should have the same shape");
   ORT_ENFORCE(!mask || mask->Shape() == X_shape, "X and mask should have the same shape");
 
-  if (ratio_value == 0.0f) {
+  const Tensor* training_mode = context->Input<Tensor>(2);
+  if (!trainable_dropout && (training_mode == nullptr || *(training_mode->Data<bool>()) == false)) {
```

Comment:
Same comment here - TrainableDropout was using `ratio == 0` to decide to go the fast-path of just memcpy.  Preserve for now?